pkg/daemon/cluster.go

```diff
@@ -32,5 +36,81 @@ type Cluster struct {
 	ConfDirHostPath string
 }
 
+type DeploymentConfig struct {
+	Name           string
+	NodeName       string
+	Type           string
+	Labels         map[string]string
+	Volumes        []v1.Volume
+	Containers     []v1.Container
+	InitContainers []v1.Container
+	*Cluster
+}
+
+const (
+	SnapshotcloneAppName = "curve-snapshotclone"
+	GrafanaAppName       = "curve-grafana"
+	EtcdAppName          = "curve-etcd"
+)
+
 func (c *Cluster) GetUUID() string { return c.UUID }
 func (c *Cluster) GetKind() string { return c.Kind }
+
+func MakeDeployment(c DeploymentConfig) (*apps.Deployment, error) {
```

Comment:
put this function in the k8sutil package and DeploymentConfig struct may not contain `Cluster`.

Reply:
done